requirements.txt

```diff
@@ -7,13 +7,13 @@ azure-core==1.21.1
 azure-identity==1.7.1
 azure-mgmt-core==1.3.0
 azure-mgmt-dns==8.0.0
-azure-mgmt-trafficmanager==0.51.0
+azure-mgmt-trafficmanager==1.0.0
```

Comment:
Oof this looks like a critical upgrade. I'll have test to this out irl to make sure everything is good since Azure SDK is notorious for backwards incompatibility.

Reply:
@ross I won't have time to test this in this week. If you want to skip just this one upgrade in this PR, everything else looks good.

Reply:
We can knock it back down for the purposes of this PR and open a new one that just bumps it for whenever someone/you get the chance to take a look at it. 

Also saw that the minor version bump when things were extracted likely caused issues on it's own, https://github.com/octodns/octodns-azure/issues/17. That's pretty annoying. I haven't had much experience with the azure python libs, but that sort of (lack of) quality was par for the course with the ruby ones. 😞